network_diffusion/utils.py

```diff
@@ -122,9 +141,11 @@ def read_tpn(
     num_of_snaps = math.ceil((max_timestamp - min_timestamp) / time_window)
 
     for snap_id in range(num_of_snaps):
-        snaps.append(get_snapshot(graph, snap_id, min_timestamp, time_window))
+        net_dict[snap_id] = get_nx_snapshot(
+            graph, snap_id, min_timestamp, time_window
+        )
 
-    return snaps
+    return net_dict
 
 
 def create_directory(dest_path: str) -> None:
```

Comment:
last remark @uniooo . Since wee added support for cogsnet with @nurek-mateusz we adapted behaviour of `TemporalNetwork` class, so your after we fix this issues your branch will have to be rebased on `develop`